src/ServiceControl.LicenseManagement/LicenseManager.cs

```diff
@@ -107,6 +108,13 @@ static string GetMachineLevelLicenseLocation()
             return LicenseFileLocationResolver.GetPathFor(Environment.SpecialFolder.CommonApplicationData);
         }
 
+        static bool MustBeNonTrialLicenseForSetup()
+        {
+            var isDockerEnvironmentVariable = Environment.GetEnvironmentVariable("SERVICECONTROL_NO_TRIAL");
```

Comment:
Isn't this confusing a bit? What we care about is that SC is running inside containers and the log error message wants to convey the message that you cannot run ServiceControl in a container with a trial license. So I don't understand why `SERVICECONTROL_NO_TRIAL` as a name is used. Because ultimately I could go and say, hmmm let's set it to false and then I can use a trial license. While when the variable is named what information it needs to convey like "SERVICECONTROL_RUNNING_IN_DOCKER" then what requirements we have in order to run with docker are more flexible and users that see that variable are less likely to tweak it. Maybe right now we are saying well we only need to ensure no trial. But maybe in the future there are other things we need to check when running in docker

So consequently if you want to keep the variable named as is you would also need to change the error message to say something like

> Cannot run ServiceControl in the current hosting environment with a trial license"

to convey the fact that it is not just about containers but actually about the license fitting a certain environment or not. Then the current name would make sense again for me. I still lean though towards expressing the intent better and make it more flexible in case we need to detect other things when running in docker